packages/vite-plugin-svelte/src/utils/preprocess.ts

```diff
@@ -194,6 +194,16 @@ export function addExtraPreprocessors(options: ResolvedOptions, config: Resolved
 	}
 }
 
+export function createSpyLangPreprocessor(map: Map<string, string>): PreprocessorGroup {
+	return {
+		script({ attributes, filename }) {
+			if (filename && attributes.lang) {
+				map.set(filename, attributes.lang as string);
```

Comment:
what happens if lang is pug? or there are a mixture of different lang attributes on module and regular script?

Reply:
Removed the preprocessor to a simple regex. If there's different lang, it would pick the first defined one as the source of truth, which should be good enough for ts support.